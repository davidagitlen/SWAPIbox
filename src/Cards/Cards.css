.Cards {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-around;
  height: 100vh;
  width: 100%;
  background: no-repeat center url(https://wallpaperplay.com/walls/full/b/e/4/124178.jpg);
  overflow: scroll;
}