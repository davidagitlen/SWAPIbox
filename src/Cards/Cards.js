import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card/Card';
import './Cards.css';

const Cards = ({data, favoriteCard, favoritesArray}) => {
  let cardsList = data.map(datum => {
    const { name } = datum;
    return <Card key={name + Date.now()} id={name + Date.now()} data={datum} favoriteCard={favoriteCard} favoritesArray={favoritesArray} />
    }
  )

  return (
    <div className='Cards'>
      {cardsList}
    </div>
  )

}


export default Cards;


Cards.propTypes = {
  data: PropTypes.array,
  favoriteCard: PropTypes.func
}