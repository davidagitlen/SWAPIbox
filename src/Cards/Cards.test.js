import React from 'react';
import { shallow } from 'enzyme';
import Cards from './Cards'

describe('Cards', () => {
  let peopleWrapper, planetsWrapper, vehiclesWrapper, favoritesWrapper, mockPeople, mockPlanets, mockVehicles, mockFavorites, favoriteCard;

  beforeEach(() => {
    mockPeople = [
      { name: 'David', homeworld: 'Kamino', species: 'wookie', population: 20},
      { name: 'Eric', homeworld: 'Tatooine', species: 'human', population: 42}
    ];
    mockPlanets = [
      { name: 'Tatooine', terrain: 'grasslands, mountains', population: 200, climate: 'temperate', residents: ['Travis', 'Robbie']},
      { name: 'Coruscant', terrain: 'desert', population: 42, climate: 'temperate', residents: ['Steve', 'Paul']}
    ]
    mockVehicles = [
      { name: 'Toyota Corolla', model: 'sedan', class: 'wheeled', passengers: 4},
      { name: 'Sand Crawler', model: 'Digger Crawler', class: 'wheeled', passengers: 30}
    ]
    mockFavorites = [
      { name: 'David', homeworld: 'Kamino', species: 'wookie', population: 20},
      { name: 'Eric', homeworld: 'Tatooine', species: 'human', population: 42},
      { name: 'Toyota Corolla', model: 'sedan', class: 'wheeled', passengers: 4}
    ];
    global.Date.now = jest.spyOn(global.Date, 'now').mockImplementation(() => 12345);
    peopleWrapper = shallow(<Cards data={mockPeople} favoriteCard={favoriteCard}/>);
    planetsWrapper = shallow(<Cards data={mockPlanets} favoriteCard={favoriteCard}/>);
    vehiclesWrapper = shallow(<Cards data={mockVehicles} favoriteCard={favoriteCard}/>);
    favoritesWrapper = shallow(<Cards data={mockFavorites} favoriteCard={favoriteCard}/>);
  });

  it('should render without crashing', () => {
    expect(peopleWrapper.length).toEqual(1)
  });

  // it('should generate a key for each Card component that is a name with Date.now()', () => {
  //   global.Date.now = jest.spyOn(global.Date, 'now').mockImplementation(() => 12345);

  //   expect(peopleWrapper.find(Card).at(0).props().id).toBe('David12345')
  // });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(peopleWrapper).toMatchSnapshot();
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(planetsWrapper).toMatchSnapshot();
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(vehiclesWrapper).toMatchSnapshot();
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(favoritesWrapper).toMatchSnapshot();
  });
});