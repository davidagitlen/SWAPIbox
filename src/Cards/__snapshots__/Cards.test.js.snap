// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards should match the snapshot with all data passed in correctly 1`] = `
<div
  className="Cards"
>
  <Card
    data={
      Object {
        "homeworld": "Kamino",
        "name": "David",
        "population": 20,
        "species": "wookie",
      }
    }
    id="David12345"
    key="David12345"
  />
  <Card
    data={
      Object {
        "homeworld": "Tatooine",
        "name": "Eric",
        "population": 42,
        "species": "human",
      }
    }
    id="Eric12345"
    key="Eric12345"
  />
</div>
`;

exports[`Cards should match the snapshot with all data passed in correctly 2`] = `
<div
  className="Cards"
>
  <Card
    data={
      Object {
        "climate": "temperate",
        "name": "Tatooine",
        "population": 200,
        "residents": Array [
          "Travis",
          "Robbie",
        ],
        "terrain": "grasslands, mountains",
      }
    }
    id="Tatooine12345"
    key="Tatooine12345"
  />
  <Card
    data={
      Object {
        "climate": "temperate",
        "name": "Coruscant",
        "population": 42,
        "residents": Array [
          "Steve",
          "Paul",
        ],
        "terrain": "desert",
      }
    }
    id="Coruscant12345"
    key="Coruscant12345"
  />
</div>
`;

exports[`Cards should match the snapshot with all data passed in correctly 3`] = `
<div
  className="Cards"
>
  <Card
    data={
      Object {
        "class": "wheeled",
        "model": "sedan",
        "name": "Toyota Corolla",
        "passengers": 4,
      }
    }
    id="Toyota Corolla12345"
    key="Toyota Corolla12345"
  />
  <Card
    data={
      Object {
        "class": "wheeled",
        "model": "Digger Crawler",
        "name": "Sand Crawler",
        "passengers": 30,
      }
    }
    id="Sand Crawler12345"
    key="Sand Crawler12345"
  />
</div>
`;

exports[`Cards should match the snapshot with all data passed in correctly 4`] = `
<div
  className="Cards"
>
  <Card
    data={
      Object {
        "homeworld": "Kamino",
        "name": "David",
        "population": 20,
        "species": "wookie",
      }
    }
    id="David12345"
    key="David12345"
  />
  <Card
    data={
      Object {
        "homeworld": "Tatooine",
        "name": "Eric",
        "population": 42,
        "species": "human",
      }
    }
    id="Eric12345"
    key="Eric12345"
  />
  <Card
    data={
      Object {
        "class": "wheeled",
        "model": "sedan",
        "name": "Toyota Corolla",
        "passengers": 4,
      }
    }
    id="Toyota Corolla12345"
    key="Toyota Corolla12345"
  />
</div>
`;
